File: 1635-number-of-good-pairs/1635-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        map<int, int> mp;
        for(int i : nums) mp[i]++;
        int ans = 0;
        for(auto& [a, b]: mp) ans += b *  (b - 1) / 2;
        return ans;
    }
};

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans = 0;
        for(int i : nums) {
            ans += map.getOrDefault(i, 0);
            map.merge(i, 1, Integer :: sum);
        }
        return ans;
    }
}